src/run.rs

```diff
@@ -539,6 +546,25 @@ fn vacuum(state: &mut State, threshold: &Byte) -> io::Result<()> {
             .then(y.ancestors.cmp(&x.ancestors))
     });
 
+    // If the user provided the keep argument, we need to filter out those images which match the
+    // provided regexes.
+    if let Some(regex_set) = keep {
+        sorted_image_nodes = sorted_image_nodes
+            .into_iter()
+            .filter(|image_node| {
+                if regex_set.is_match(&image_node.image_info.repository_tag) {
+                    info!(
+                        "Skipping image {} which matches one or more of the provided regexs",
+                        image_node.image_info.repository_tag
+                    );
```

Comment:
How about:

```rust
                    info!(
                        "Ignored image {} due to the {} flag.",
                        image_node.image_info.repository_tag.code_str(),
                        "--keep".code_str(),
                    );
```

(There was missing punctuation and a missing trailing comma, and I think it is good to reference the flag that caused the image to be skipped. Also "regexs" seemed like a strange pluralization to me.)